backend/src/entities/member.entity.ts

```diff
@@ -0,0 +1,193 @@
+import {
+    Entity,
+    Column,
+    ManyToOne,
+    OneToMany,
+    ManyToMany,
+    JoinColumn,
+    JoinTable,
+    PrimaryGeneratedColumn,
+    CreateDateColumn,
+    UpdateDateColumn,
+    DeleteDateColumn,
+  } from 'typeorm';
+import { MemberType } from './memberType.entity';
+import { Ministry } from './ministry.entity';
+import { ContactLog } from './contactLog.entity';
+import { MemberUnderMinister } from './memberUnderMinister.entity';
+import { MemberMinistry } from './memberMinistry.entity';
+  
+  export enum MaritalStatus {
+    Married = 'Married',
+    Single = 'Single',
+    Divorced = 'Divorced',
+    Widowed = 'Widowed',
+  }
+
+  export enum Gender {
+    Male = 'Male',
+    Female = 'Female',
+  }
+
+  @Entity()
+  export class Member {
+    @PrimaryGeneratedColumn()
+    id: number;
+  
+    @ManyToOne(() => MemberType, (memberType) => memberType.members)
+    @JoinColumn({ name: 'member_type_id' })
+    memberType: MemberType;
+
+    @OneToMany(() => MemberMinistry, (memberMinistry) => memberMinistry.member)
+    ministries: MemberMinistry[];
+
+    @Column({
+      name: 'first_name',
+    })
+    firstName: string;
+
+    @Column({
+      name: 'middle_name',
+      nullable: true,
+    })
+    middleName: string;
+  
+    @Column({
+      name: 'last_name',
+    })
+    lastName: string;
+  
+    @Column({
+      unique: true,
+      nullable: false,
+    })
+    username: string;
+
+    @Column({
+      name: 'phone_number',
+    })
+    phoneNumber: string;
+
+    @Column({ nullable: true })
+    email: string;
+
+    @Column({ nullable: true })
+    address: string;
+
+    @Column({
+      name: 'previous_church',
+      nullable: true,
+    })
+    previousChurch: string;
+
+    @Column({
+      name: 'role_in_previous_church',
+      nullable: true,
+    })
+    roleInPreviousChurch: string;
+
+    @Column({ name: 'is_baptised' })
+    isBaptised: string;
+
+    @Column({
+      name: 'spouse_name',
+      nullable: true,
+    })
+    spouseName: string;
+
+    @Column({
+      name: 'children_names',
+      nullable: true,
+    })
+    childrenNames: string;
+
+    @Column({
+      name: 'emergency_contact',
+      nullable: true,
+    })
+    emergencyContact: string;
+
+    @Column({
+      name: 'marital_status',
+      type: 'enum',
+      enum: MaritalStatus,
+    })
+    maritalStatus: MaritalStatus;
+
+    @Column({
+      name: 'has_letter_from_prev_church',
+      nullable: true,
```

Comment:
why is this nullable? It is either True or False, it can be False by default. I don't think we have mroe options to this (የተቀደደበት).